lib/auth/tun.go

```diff
@@ -690,15 +712,14 @@ func (c *TunClient) fetchAuthServers() ([]utils.NetAddr, error) {
 
 // getAuthServers returns a sorted list of auth servers
 func (c *TunClient) getAuthServers() (out []utils.NetAddr) {
-	// protected clone into 'out':
-	func() {
-		c.Lock()
-		defer c.Unlock()
-		out = make([]utils.NetAddr, len(c.authServers))
-		copy(out, c.authServers)
-	}()
-	// sort & return:
-	sort.Sort(byAddress(out))
+	c.Lock()
+	defer c.Unlock()
+
+	// return static auth servers followed by discovered ones. this guarantees
+	// that the client will try statically configured ones first
+	out = make([]utils.NetAddr, 0, len(c.staticAuthServers)+len(c.discoveredAuthServers))
```

Comment:
I really worry now that this `Static` vs non `Static` code will plague code everywhere. There should be a backend abstraction that simply joins statically configured and dynamically configured results everywhere.